tools/tfm_debug_symbol_list.yaml

```
@@ -0,0 +1,18 @@
+# This is is just a list for the secure veneer functions, the generated file 
+# can be used to add linker instruction to not remove them, so debug info can
+# be provided 
+
+{
+  "name": "TF-M generated veneer symbols",
+  "type": "generated_file_list",
+  "version_major": 0,
+  "version_minor": 1,
+  "file_list": [
+    {
+        "name": "Secure Veneers symbols file",
+        "short_name": "tfm_veneers_c",
+        "template": "tools/templates/veneer_debug_symbols.template",
```

```suggestion
        "template": "tools/templates/veneer_function_names.template",
```

We can be more specific than "symbols" and (IMHO) since we remove --undefined, we don't need to mention debug specifically as that is only related to how this file is used, and not related to what this file is.

---

Let's replace everywhere in this PR "symbol" with function_name to be more specific.